swarm/src/lib.rs

```diff
@@ -514,7 +514,7 @@ where
                     };
                     match dial {
                         Ok(fut) => fut
-                            .map(|r| (address, r.map_err(TransportError::Other)))
+                            .map(|r| (address, r.map_err(|e| TransportError::Other(e))))
```

Comment:
Why not stick to passing the variant as a function reference?

Reply:
```suggestion
                            .map(|r| (address, r.map_err(TransportError::Other)))
```